src/main/java/uk/gov/pay/api/json/RequestJsonParser.java

```diff
@@ -41,136 +43,123 @@
-        }
-        return fieldValue;
+    private static Boolean validateAndGetDelayedCapture(JsonNode paymentRequest) {
+        return getValue(
+                paymentRequest.get(DELAYED_CAPTURE_FIELD_NAME),
+                aPaymentError(DELAYED_CAPTURE_FIELD_NAME, CREATE_PAYMENT_VALIDATION_ERROR, "Must be true or false"),
+                aPaymentError(DELAYED_CAPTURE_FIELD_NAME, CREATE_PAYMENT_VALIDATION_ERROR, "Must be true or false"),
+                JsonNode::isBoolean,
+                JsonNode::booleanValue);
     }
 
-    private static Integer getIntegerValue(JsonNode fieldNode, PaymentError formatError) {
-        Integer fieldValue = null;
-        if (notNull(fieldNode)) {
-            check(fieldNode.isInt(), formatError);
-            fieldValue = fieldNode.intValue();
-        }
-        return fieldValue;
+    private static Integer validateAndGetAmount(JsonNode paymentRequest, Code validationError, Code missingError) {
+        return getValue(
+                paymentRequest.get(AMOUNT_FIELD_NAME),
+                aPaymentError(AMOUNT_FIELD_NAME, validationError, "Must be a valid numeric format"),
+                aPaymentError(AMOUNT_FIELD_NAME, missingError),
+                JsonNode::isInt,
+                JsonNode::intValue);
     }
 
-    private static Boolean getBooleanValue(JsonNode fieldNode, PaymentError formatError) {
-        Boolean fieldValue = null;
-        if (notNull(fieldNode)) {
-            check(fieldNode.isBoolean(), formatError);
-            fieldValue = fieldNode.booleanValue();
+    private static ExternalMetadata validateAndGetMetadata(JsonNode paymentRequest) {
+        Map<String, Object> metadataMap = null;
+        try {
+            metadataMap = objectMapper.convertValue(paymentRequest.get("metadata"), Map.class);
+        } catch (IllegalArgumentException e) {
+            PaymentError paymentError = new PaymentError(METADATA, CREATE_PAYMENT_METADATA_VALIDATION_ERROR,
+                    "Field [metadata] must be an object of JSON key-value pairs");
+            throw new WebApplicationException(Response.status(SC_UNPROCESSABLE_ENTITY).entity(paymentError).build());
+        }
+        ExternalMetadata metadata = new ExternalMetadata(metadataMap);
+        Set<ConstraintViolation<ExternalMetadata>> violations = validator.validate(metadata);
+        if (violations.size() > 0) {
+            String message = violations.stream().map(v -> v.getMessage()).collect(Collectors.joining(";\n"));
+            PaymentError paymentError = new PaymentError(METADATA, CREATE_PAYMENT_METADATA_VALIDATION_ERROR, message);
+            throw new WebApplicationException(Response.status(SC_UNPROCESSABLE_ENTITY).entity(paymentError).build());
         }
-        return fieldValue;
+        return metadata;
     }
-
-    private static boolean notNull(JsonNode fieldNode) {
-        return fieldNode != null && !fieldNode.isNull();
+    
+    private static <T> T getValue(JsonNode jsonNode,
```

Comment:
Perhaps calling this `validateAndGetValue` might be more descriptive and in-line with other private methods since it is validating as part of return the value.